index.js

```diff
@@ -42,23 +45,23 @@ module.exports = function setHeader(res, name, value) {
   // a private `_headers` object. We're going to override the value that got set
   // using the Object.defineProperty so nobody can set the same header again.
   //
-  Object.defineProperty(res._headers, key, {
+  Object.defineProperty(res[symbol], key, {
     configurable: false,
     enumerable: true,
 
     //
     // Return the value that got set using our `setHeader` method.
     //
     get: function get() {
-      return value;
+      return (symbols.length == 1 ? [ key, value ] : value);
     },
 
     //
     // Log an override attempt on our protected header.
     //
     set: function set(val) {
       debug('attempt to override header %s:%s with %s', name, value, val);
-      return value;
+      return (symbols.length == 1 ? [ key, value ] : value);
```

Comment:
I'd keep only `value` here, `[ key, value ]` is an implementation detail that is not important here.

Reply:
I don't know if it breaks anything but lgtm